File: frontend/src/types/notification.ts
export type NotificationType = "reminder" | "info" | "warning";

export type DeliveryChannel = "in_app" | "email" | "both";

export interface Notification {
  id: string;
  message: string;
  type: NotificationType;
  is_read: boolean;
  conversation_id: string | null;
  created_at: string;
}

export interface NotificationPreferences {
  delivery_channel: DeliveryChannel;
  quiet_hours_start: string | null;
  quiet_hours_end: string | null;
  quiet_hours_timezone: string;
}

export interface NotificationPreferencesUpdate {
  delivery_channel?: DeliveryChannel;
  quiet_hours_start?: string | null;
  quiet_hours_end?: string | null;
  quiet_hours_timezone?: string;
}

export interface PaginatedNotifications {
  items: Notification[];
  total: number;
  limit: number;
  offset: number;
}

export interface UnreadCountResponse {
  count: number;
}

File: src/models/notification.py
"""Notification models for Feature 010."""

from datetime import datetime, time
from enum import Enum
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, Field


class NotificationType(str, Enum):
    """Valid notification types."""

    REMINDER = "reminder"
    INFO = "info"
    WARNING = "warning"


class DeliveryChannel(str, Enum):
    """Notification delivery channel options."""

    IN_APP = "in_app"
    EMAIL = "email"
    BOTH = "both"


class Notification(BaseModel):
    """A notification from the agent to a user."""

    id: UUID
    user_id: UUID
    conversation_id: Optional[UUID] = None
    message: str
    type: NotificationType
    is_read: bool = False
    created_at: datetime


class NotificationPreferences(BaseModel):
    """Per-user notification delivery settings."""

    delivery_channel: DeliveryChannel = DeliveryChannel.IN_APP
    quiet_hours_start: Optional[time] = None
    quiet_hours_end: Optional[time] = None
    quiet_hours_timezone: Optional[str] = "UTC"


class NotificationPreferencesUpdate(BaseModel):
    """Request model for updating notification preferences."""

    delivery_channel: Optional[DeliveryChannel] = None
    quiet_hours_start: Optional[time] = None
    quiet_hours_end: Optional[time] = None
    quiet_hours_timezone: Optional[str] = None


class CreateNotificationRequest(BaseModel):
    """Validated input for creating a notification."""

    message: str = Field(..., min_length=1, max_length=500)
    type: NotificationType = NotificationType.INFO
